models/utilities/default_model_mod.f90

```diff
@@ -283,6 +300,162 @@ subroutine pert_model_copies(state_ens_handle, ens_size, pert_amp, interf_provid
+do ivar = 1, MAX_STATE_VARIABLES
+   ! If the first element in the row is an empty string, the loop has exceeded the extent of the variables
+   if (nml_state_vars(5*ivar-4) == '') then
+      state_vars%nvars = ivar-1
+      exit
+   endif
+enddo
+
+! Allocate the arrays in the var derived type
+allocate(state_vars%netcdf_var_names(state_vars%nvars), state_vars%qtys(state_vars%nvars), state_vars%clamp_values(state_vars%nvars, 2), state_vars%updates(state_vars%nvars))
+
+RowsLoop : do i = 1, state_vars%nvars
+
+   netcdf_var_name = trim(nml_state_vars(5*i-4))
+   state_vars%netcdf_var_names(i) = trim(netcdf_var_name)
+
+   dart_qty_str = trim(nml_state_vars(5*i-3))
+   call to_upper(dart_qty_str)
+   ! Make sure DART qty is valid
+   state_vars%qtys(i) = get_index_for_quantity(dart_qty_str)
+   if( state_vars%qtys(i) < 0 ) then
+      write(string1,*) 'The quantity specified in the &model_nml "', dart_qty_str, '", is not present in obs_kind_mod.f90'
+      call error_handler(E_ERR,'get_state_variables_clamp',string1)
+   endif
+
+   if (nml_state_vars(5*i-2) /= 'NA') then
+      read(nml_state_vars(5*i-2), '(d16.8)') state_vars%clamp_values(i,1)
+   else
+      state_vars%clamp_values(i,1) = MISSING_R8
+   endif
+
+   if (nml_state_vars(5*i-1) /= 'NA') then
+      read(nml_state_vars(5*i-1), '(d16.8)') state_vars%clamp_values(i,2)
+   else
+      state_vars%clamp_values(i,2) = MISSING_R8
+   endif
+
+   update = trim(nml_state_vars(5*i))
+   call to_upper(update)
+   select case (update)
+      case ('UPDATE')
+         state_vars%updates(i) = .true.
+      case ('NO_COPY_BACK')
+         state_vars%updates(i) = .false.
+      case default
+         write(string1,'(A)')  'Invalid update variable in &model_nml:model_state_variable - only UPDATE or NO_COPY_BACK are supported'
+         write(string2,'(6A)') 'Issue: ', trim(netcdf_var_name), ', ', trim(dart_qty_str), ', ', trim(update)
+         call error_handler(E_ERR,'get_state_variables_clamp',string1, text2=string2)
+   end select
+
+   ! Checking that the rows in the nml entry are all complete
+   if ( dart_qty_str == '' .or. nml_state_vars(5*i-2) == '' .or. nml_state_vars(5*i-1) == '' .or. update == '' ) then
+      string1 = 'model_nml:model_state_variables not fully specified'
+      call error_handler(E_ERR, 'get_state_variables_clamp', string1)
+   endif
+
+enddo RowsLoop
+
+end subroutine get_state_variables_clamp
+
+!--------------------------------------------------------------------
+
+!> Reads in model_nml:model_state_variables and returns a
+!> state_var_type state_vars with nvars ; netcdf variable names ;
+!> qtys (kinds) ; updates
+!
+!> Verifies that the namelist was filled in correctly, and checks
+!> that there are valid entries for the dart_kind.
+
+subroutine get_state_variables_noclamp(nml_state_vars, MAX_STATE_VARIABLES, state_vars)
+
+character(len=*),   intent(inout) :: nml_state_vars(:)
```

Comment:
nitpick: not a namelist, only an character array

Reply:
Yeah I guess my thought here was that the character array is obtained directly from the nml, so that's why I added that - I'll change it

Reply:
this is a utility routine, so you shouldn't predict or constrain how the calling code chooses to use it.  a model_mod could read a character array out of a config file that wasn't the namelist file, or construct one by an interactive script, couldn't it?